### PR TITLE
Add warnings of non existent periods

### LINKED ISSUE #29121: MA warnings for periods that are not present
The new grouping tab in MA allows users to explicitly state what periods are present. It is now possible to request a period that may not exist for some of the data. This is to give the user some feedback.

1. If you have say 5 runs and the first 4 have 2 periods. Then the second period will be in orange (to signify some of the data does not have a 2nd period).

2. If you have 5 runs with one period. Then asking for period 2 will cause it to be in red

We could add a tooltip or something to give more detailed feedback.

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**

This adds warnings to the Muon Analysis GUI when one or more runs do not contain all the relevant periods to calculate a group fully. If some runs are missing periods the rows are colored in yellow and if all runs are missing periods they are colored in red. The tooltips are also updated to reflect this.

The names of groups calculated have also been updated to remove the periods information as this is now contained in the group.

**To test:**
You will need access to the ISIS data archive:
  * Load in a run with single period data and one with several periods. HIFI84447 and HIFI44447 should suffice for this.
  * Check that warnings are shown for all groups containing a period that is not 1.
  * Check that is you switch to just 44447 the yellow rows become red as they are now invalid for all groups.
  * Check that if you switch to just 84447 all the rows clear.

Fixes #29121 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
could we have the colours on the pairs too?

File: docs/source/release/v5.1.0/muon.rst [hunk 1]
```diff
@@ -56,6 +56,7 @@ Improvements
 - The ALC interface in workbench will now show errors by default. The error bars can also be turned on/off using the right-click plot menu.
 - Have updated the FDA GUI so that it functions correctly for frequency transforms and single fits.
 - Added in default group and pair selection when loading grouping files from xml.
+- Added warnings when not all selected runs have the relevant periods to calculate certain groups.
```
Comment (AnthonyLim23) [inline docs/source/release/v5.1.0/muon.rst:59]:
```suggestion
- Added warnings when the selected runs do not have the relevant periods to calculate the groups or pairs.
```

File: scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py [hunk 2]
```diff
@@ -14,6 +14,21 @@
 maximum_number_of_groups = 20
 
 
+class RowValid(Enum):
+    invalid_for_all_runs = 0
+    valid_for_all_runs = 2
+    valid_for_some_runs = 1
+
+
+row_colors = {RowValid.invalid_for_all_runs: (255, 0, 0), RowValid.valid_for_some_runs: (255, 255, 0),
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py:23]: are these numbers just the colours?

File: scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_view.py [hunk 3]
```diff
@@ -404,3 +386,12 @@ def get_group_range(self):
     def set_group_range(self, range):
         self.group_range_min.setText(range[0])
         self.group_range_max.setText(range[1])
+
+    # def change_row_color(self, color, row_index):
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_view.py:390]: is this supposed to be here?

File: scripts/test/Muon/grouping_tab/grouping_table_presenter_test.py [hunk 4]
```diff
@@ -402,6 +402,36 @@ def test_when_adding_group_to_empty_table_it_is_selected(self):
 
         self.assertEquals(self.model.selected_groups, ['group_1'])
 
+    def test_update_view_from_model_correctly_adds_warnings_for_invalid_periods(self):
+        self.presenter.validate_periods_list = mock.MagicMock(return_value=RowValid.invalid_for_all_runs)
+        self.presenter._view.add_entry_to_table = mock.MagicMock()
+        self.presenter.add_group_to_model(MuonGroup('group_1', [1,2,3,4], [3]))
```
Comment (AnthonyLim23) [inline scripts/test/Muon/grouping_tab/grouping_table_presenter_test.py:408]: can we have some explanation for what the numbers mean?

File: scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py (hunk 2 above)
Comment (Mantid-Matthew) [reply scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py:23]: Yes these are the colours specified in RGB. I'll add a comment to clarify which of the various colour specification schemes is being used here. I have a vague memory of Aidy not wanting yellow and red for the warning error colours here. Does that ring any bells with you?

File: scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_view.py (hunk 3 above)
Comment (Mantid-Matthew) [reply scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_view.py:390]: No leftover comment.

File: scripts/test/Muon/grouping_tab/grouping_table_presenter_test.py (hunk 4 above)
Comment (Mantid-Matthew) [reply scripts/test/Muon/grouping_tab/grouping_table_presenter_test.py:408]: I'll switch to the below to clarify things.
```
self.presenter.add_group_to_model(MuonGroup(name='group_1', detectors=[1,2,3,4], periods=[3]))
```

File: scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py (hunk 2 above)
Comment (AnthonyLim23) [reply scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py:23]: yes it does. The colours looked ok to me. It was more the magic numbers that bothered me

**Comment by Mantid-Matthew** [comment]
> could we have the colours on the pairs too?

This isn't an enormously difficult feature but does require some reorganizing of where the validity of periods is stored and as always when new state is added some care taken to ensure it updates correctly. Given we are approaching the end of the sprint and my time on the project I'd like to push back slightly and suggest this foes in as is and a new issue is created propagate the warnings onto relevant periods.

**Comment by Mantid-Matthew** [comment]
Had more time than anticipated to have added the colors to the pair table as well.

**Review by AnthonyLim23 — APPROVED** [review]

**Pull request merged**